pycls/core/trainer.py

```diff
@@ -60,18 +123,44 @@ def setup_model():
     """Sets up a model for training or testing and log the results."""
     # Build the model
     model = builders.build_model()
+    ema_model = builders.build_model()
+
+    if cfg.FSDP.ENABLED:
+        fsdp_config = {}
+        fsdp_config["mixed_precision"] = cfg.FSDP.MIXED_PRECISION
```

Comment:
Should this be `True` if we set TRAIN.MIXED_PRECISION = True?

If so, should we reuse that config variable instead of definition FSDP.MIXED_PRECISION

Reply:
If TRAIN.MIXED_PRECISION == True and FSDP.MIXED_PRECISION == False we will be using full FP16. If TRAIN.MIXED_PRECISION = False in this setting, inputs and labels will not be converted to FP16 automatically.

Reply:
In v1, we can just reuse TRAIN.MIXED_PRECISION. That will be simpler to explain to the researchers.

1. If TRAIN.MIXED_PRECISION is false then everything runs in fp32
2. If TRAIN.MIXED_PRECISION is true and FSDP is false then it defaults to AMP O1.
3. If TRAIN.MIXED_PRECISION is true and FSDP is true then it defaults to AMP O2 with FP32 wrapping of Layer Norm.